stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -0,0 +1,1150 @@
+
+    let alice_account = get_account(&mut peer, &key_to_stacks_addr(&alice).into());
+    assert_eq!(alice_account.stx_balance.amount_unlocked(), INITIAL_BALANCE);
+    assert_eq!(alice_account.stx_balance.amount_locked(), 0);
+    assert_eq!(alice_account.stx_balance.unlock_height(), 0);
+
+    // next tenure include Alice's lockup
+    let tip = get_tip(peer.sortdb.as_ref());
+    let alice_lockup = make_pox_lockup(
+        &alice,
+        0,
+        ALICE_LOCKUP,
+        AddressHashMode::SerializeP2PKH,
+        key_to_stacks_addr(&alice).bytes,
+        4,
+        tip.block_height,
+    );
+
+    let tip_index_block = peer.tenure_with_txs(&[alice_lockup], &mut coinbase_nonce);
+
+    // check the stacking minimum
+    let total_liquid_ustx = get_liquid_ustx(&mut peer);
+    let min_ustx = with_sortdb(&mut peer, |chainstate, sortdb| {
+        chainstate.get_stacking_minimum(sortdb, &tip_index_block)
+    })
+    .unwrap();
+    assert_eq!(
+        min_ustx,
+        total_liquid_ustx / POX_TESTNET_STACKING_THRESHOLD_25
+    );
+
+    // no reward addresses
+    let reward_addrs = with_sortdb(&mut peer, |chainstate, sortdb| {
+        get_reward_addresses_with_par_tip(chainstate, &burnchain, sortdb, &tip_index_block)
+    })
+    .unwrap();
+    assert_eq!(reward_addrs.len(), 0);
+
+    // check the first reward cycle when Alice's tokens get stacked
+    let tip_burn_block_height = get_par_burn_block_height(peer.chainstate(), &tip_index_block);
+    let alice_first_reward_cycle = 1 + burnchain
+        .block_height_to_reward_cycle(tip_burn_block_height)
+        .unwrap();
+
+    assert_eq!(
+        alice_first_reward_cycle as u128,
+        EXPECTED_ALICE_FIRST_REWARD_CYCLE
+    );
+    let height_target = burnchain.reward_cycle_to_block_height(alice_first_reward_cycle) + 1;
+
+    // alice locked, so balance should be 0
+    let alice_balance = get_balance(&mut peer, &key_to_stacks_addr(&alice).into());
+    assert_eq!(alice_balance, 0);
+
+    while get_tip(peer.sortdb.as_ref()).block_height < height_target {
+        latest_block = peer.tenure_with_txs(&[], &mut coinbase_nonce);
+    }
+
+    // produce blocks until epoch 2.1
+    while get_tip(peer.sortdb.as_ref()).block_height < epochs[3].start_height {
+        peer.tenure_with_txs(&[], &mut coinbase_nonce);
+        alice_rewards_to_v2_start_checks(latest_block, &mut peer);
+    }
+
+    // in the next tenure, PoX 2 should now exist.
+    // Lets have Bob lock up for v2
+    // this will lock for cycles 8, 9, 10
+    //  the first v2 cycle will be 8
+    let tip = get_tip(peer.sortdb.as_ref());
+
+    let bob_lockup = make_pox_2_lockup(
+        &bob,
+        0,
+        BOB_LOCKUP,
+        PoxAddress::from_legacy(
+            AddressHashMode::SerializeP2PKH,
+            key_to_stacks_addr(&bob).bytes,
+        ),
+        3,
+        tip.block_height,
+    );
+
+    // Alice _will_ auto-unlock: she can stack-extend in PoX v2
```

Comment:
Shouldn't this be `Alice _will not_ auto-unlock`?  Also, wouldn't Alice have unlocked from pox-2 activating?

Reply:
No -- the comment is confusing, but "correct". Alice stacked in pox v1 and so would have experienced auto-unlock but instead she stack-extends in PoX v2.

The v2 unlock height configuration setting _doesn't_ need to match epoch2.1 start (and it doesn't in these tests) -- the auto-unlock occurs after pox-2 is activated, which is why Alice can extend: she's currently locked, and she's extending so she doesn't miss a cycle.